.github/workflows/semantic-release.yaml

```diff
@@ -1,32 +1,82 @@
 name: Semantic Release
 
 on:
-  push:
-    branches:
-      - staging
-      - production
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: 'Environment to release to. Select "staging" for a pre-release or "production" for a full release.'
+        options:
+          - staging
+          - production
+        required: true
 
 jobs:
+  # Create a release
   create-release:
     runs-on: ubuntu-latest
+    outputs:
+      published: ${{ steps.release.outputs.published }}
+      tag: ${{ steps.release.outputs.tag }}
     steps:
-      - name: checkout-repo
+      - name: Checkout repo
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
           fetch-tags: true
           persist-credentials: false
           token: ${{ secrets.KIVA_ROBOT_GITHUB_PAT }}
-      - name: checkout-actions
+      - name: Checkout actions
         uses: actions/checkout@v4
         with:
           repository: kiva/github-actions
           token: ${{ secrets.KIVA_ROBOT_GITHUB_PAT }}
           persist-credentials: false
           path: .github/
-      - name: semantic-release
+      - name: Run semantic-release
+        id: release
         uses: ./.github/actions/semantic-release
         env:
           GITHUB_PAT: ${{ secrets.KIVA_ROBOT_GITHUB_PAT }}
         with:
           config: "node"
+
+  # Build and push the image to ECR if the release was published
+  build-image:
```

Comment:
Can most of this code move back into the gh-actions repo at some point for easier reuse?

Reply:
Yeah, gha allows for reusable/shareable workflows. I think it would be great if we could eventually setup these workflows in the github actions repo and then reference them in all the app repos.